middleware/src/middleware.go

```diff
@@ -597,3 +599,38 @@ func (middleware *Middleware) SetRootPassword(args rpcmessages.SetRootPasswordAr
 		Code:    rpcmessages.ErrorSetRootPasswordTooShort,
 	}
 }
+
+// BaseVersionForVersionEndpoint supplys the Base version for the middleware REST endpoint `/version`.
+// The version is primarily obtained via a Redis GET.
+// If Redis can not be reached, the version is read from disk.
+// If both redis and the disk can not supply a version, then "0.0.0" is returned.
+// **This is not a RPC endpoint function.**
+func (middleware *Middleware) BaseVersionForVersionEndpoint() string {
+	//const versionFilePath string = "/opt/shift/config/version"
+	const versionFilePath string = "/tmp/version"
+	const fallbackVersion string = "0.0.0"
+
+	version, err := middleware.redisClient.GetString("base:version") // TODO: have this as a const somewhere
```

Comment:
`TODO` could be resolved with something as discussed here: https://github.com/digitalbitbox/bitbox-base/pull/218#discussion_r330447544

Reply:
Maybe I am lacking some context here, but why should be const? It's typically only constant at runtime, but not at compile time.

Reply:
Oh sorry, that TODO is not clear. 

I thought about having the Redis keys (e.g. `base:version`) as a const in something like `redis/keys.go` in https://github.com/digitalbitbox/bitbox-base/pull/218#discussion_r330447544

Reply:
Ah, I see.